src/Microsoft.Azure.SignalR/ServerConnections/ServiceConnection.cs

```diff
@@ -113,6 +114,17 @@ private async Task ProcessOutgoingMessagesAsync(ServiceConnectionContext connect
                     var buffer = result.Buffer;
                     if (!buffer.IsEmpty)
                     {
+                        // We assume the first response message would be a HandshakeResponse.
+                        if (connection.IsMigrated)
+                        {
+                            if (SignalRProtocol.HandshakeProtocol.TryParseResponseMessage(ref buffer, out var message))
+                            {
+                                connection.IsMigrated = false;
+                                connection.Application.Input.AdvanceTo(buffer.End);
```

Comment:
should be `buffer.Start`

Reply:
Why? We just want to skip this message.

Reply:
`buffer.End` skips all the content inside the pipe, for example "{handshake} {real message}"...

---
In reply to: [347766993](https://github.com/Azure/azure-signalr/pull/739#discussion_r347766993) [](ancestors = 347766993)

Reply:
I'm afraid our existing logic will face the same problem.
Maybe we have to fix them all if this is very important.
( since only ping and close message will in this pipe I don't think it is necessary